app/services/candidate_interface/find_changed_apply_again_applications.rb

```diff
@@ -0,0 +1,54 @@
+module CandidateInterface
+  class FindChangedApplyAgainApplications
+    def call
+      apply_again_forms.find_each.lazy.select do |application_form|
+        changed?(application_form)
+      end
+    end
+
+  private
+
+    def apply_again_forms
+      ApplicationForm
+        .where(
+          phase: 'apply_2',
+          recruitment_cycle_year: RecruitmentCycle.current_year,
+        )
+        .includes(previous_application_form: [:application_qualifications], application_qualifications: [])
```

Comment:
Nit, but could this alternatively be:

`.includes(:application_qualifications, previous_application_form: [:application_qualifications])`

Reply:
:+1: that's much better